core/src/main/java/com/datastax/oss/driver/api/core/type/codec/TypeCodec.java

```diff
@@ -184,7 +190,8 @@ default boolean accepts(DataType cqlType) {
    * If you choose not to implement this method, don't throw an exception but instead return a
    * constant string (for example "XxxCodec.format not implemented").
    */
-  String format(T value);
+  @Nullable
+  String format(@Nullable T value);
```

Comment:
All built-in codec return a non-null value, and the request logger currently fails if this method returns null. I think it's fair to make it `@NonNull`.